ergo-lib/src/chain/json/transaction.rs

```diff
@@ -18,6 +18,21 @@ pub struct TransactionJson {
     pub outputs: Vec<ErgoBox>,
 }
 
+#[derive(Serialize, Deserialize, PartialEq, Debug, Clone)]
+pub struct UnsignedTransactionJson {
+    /// unsigned inputs, that will be spent by this transaction.
+    #[cfg_attr(feature = "json", serde(rename = "inputs"))]
+    pub inputs: Vec<UnsignedInput>,
+    /// inputs, that are not going to be spent by transaction, but will be reachable from inputs
+    /// scripts. `dataInputs` scripts will not be executed, thus their scripts costs are not
+    /// included in transaction cost and they do not contain spending proofs.
+    #[cfg_attr(feature = "json", serde(rename = "dataInputs"))]
+    pub data_inputs: Vec<DataInput>,
+    /// box candidates to be created by this transaction
+    #[cfg_attr(feature = "json", serde(rename = "outputs"))]
+    pub outputs: Vec<ErgoBoxCandidate>,
+}
+
 #[cfg(test)]
 mod tests {
     use crate::chain::transaction::unsigned::UnsignedTransaction;
```

Comment:
Curiously, the test `unsigned_tx_roundtrip()` passes beforehand, but when we tried to actually deserialize from JSON an unsigned transaction (from the wasm bindings but those use the same `serde_json::from_str()` method) it complains that there's a missing field `tx_id`.